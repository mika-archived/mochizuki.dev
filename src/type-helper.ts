import { StyledComponent } from "styled-components";

// Extract Props of React Components
export type ExtractProps<T> =
  // prettier-ignore
  T extends StyledComponent<any, any, infer R, never> ? R
  : T extends React.FunctionComponent<infer R> ? R
  : string;

export type ReactClassName = {
  className?: string;
};
