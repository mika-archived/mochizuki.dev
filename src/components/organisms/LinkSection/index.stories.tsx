import React from "react";
import { storiesOf } from "@storybook/react";

import { ExtractProps } from "../../../type-helper";

import LinkSection from ".";

const links: ExtractProps<typeof LinkSection>["links"] = [
  { account: "MikazukiFuyuno", label: "Twitter", href: "/", icon: "twitter", prefix: "brands" },
  { account: "me@mochizuki.moe", label: "Email", href: "/", icon: "envelope", prefix: "light" },
  { account: "MikazukiFuyuno", label: "Twitter", href: "/", icon: "twitter", prefix: "brands" },
  { account: "me@mochizuki.moe", label: "Email", href: "/", icon: "envelope", prefix: "light" },
  { account: "MikazukiFuyuno", label: "Twitter", href: "/", icon: "twitter", prefix: "brands" },
  { account: "me@mochizuki.moe", label: "Email", href: "/", icon: "envelope", prefix: "light" }
];

storiesOf("organisms/LinkSection", module).add("default", () => (
  <>
    <LinkSection links={links} />
  </>
));
