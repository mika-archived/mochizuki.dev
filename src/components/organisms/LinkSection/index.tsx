import React from "react";
import styled from "styled-components";

import FontAwesome from "../../atoms/FontAwesome";
import { Grid, Row } from "../../atoms/GridLayout";
import { Heading2 } from "../../atoms/Headings";
import BrandLinkGroup from "../../molecules/BrandLinkGroup";
import NormalLinkGroup from "../../molecules/NormalLinkGroup";

import { ExtractProps } from "../../../type-helper";

type Props = {
  links: (ExtractProps<typeof BrandLinkGroup> & { prefix: ExtractProps<typeof FontAwesome>["prefix"] })[];
};

const LinkGroup = styled.div`
  margin-bottom: 16px;
`;

const LinkSection: React.FC<Props> = ({ links }) => {
  return (
    <section>
      <Heading2>Links</Heading2>
      <Grid rows={6}>
        {links.map(w => (
          <Row key={`${w.icon}-${w.label}`} sm={3} md={2}>
            <LinkGroup>
              {w.prefix === "brands" ? (
                <BrandLinkGroup account={w.account} label={w.label} href={w.href} icon={w.icon} />
              ) : (
                <NormalLinkGroup account={w.account} label={w.label} href={w.href} icon={w.icon} />
              )}
            </LinkGroup>
          </Row>
        ))}
      </Grid>
    </section>
  );
};

export default LinkSection;
