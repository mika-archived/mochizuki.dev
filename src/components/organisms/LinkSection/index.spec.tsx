import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, render } from "@testing-library/react";

import { ExtractProps } from "../../../type-helper";

import LinkSection from ".";

const links: ExtractProps<typeof LinkSection>["links"] = [
  { account: "MikazukiFuyuno", label: "Twitter 1", href: "/", icon: "twitter", prefix: "brands" },
  { account: "me@mochizuki.moe", label: "Email 1", href: "/", icon: "envelope", prefix: "light" },
  { account: "MikazukiFuyuno", label: "Twitter 2", href: "/", icon: "twitter", prefix: "brands" },
  { account: "me@mochizuki.moe", label: "Email 2", href: "/", icon: "envelope", prefix: "light" },
  { account: "MikazukiFuyuno", label: "Twitter 3", href: "/", icon: "twitter", prefix: "brands" },
  { account: "me@mochizuki.moe", label: "Email 3", href: "/", icon: "envelope", prefix: "light" }
];

afterEach(cleanup);

test("does not have changes from snapshot", () => {
  const { asFragment } = render(<LinkSection links={links} />);

  expect(asFragment()).toMatchSnapshot();
});
