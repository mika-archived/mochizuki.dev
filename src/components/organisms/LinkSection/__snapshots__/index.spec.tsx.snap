// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes from snapshot 1`] = `
<DocumentFragment>
  .c1 {
  display: inline-block;
}

.c0 {
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 21px;
  font-weight: normal;
}

.c8 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 14px;
  overflow: hidden;
  color: #888;
  text-overflow: ellipsis;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.c7 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 14px;
  overflow: hidden;
  color: #333;
  text-overflow: ellipsis;
}

.c5 {
  font-size: 36px;
  line-height: 1.2;
  color: #333;
}

.c6 {
  width: 100%;
  margin-left: 6px;
}

.c14 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 14px;
  overflow: hidden;
  color: #888;
  text-overflow: ellipsis;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.c13 {
  margin: 0;
  font-family: "游ゴシック体","Yu Gothic",YuGothic,"ヒラギノ角ゴシック Pro","Hiragino Kaku Gothic Pro",メイリオ,Meiryo,sans-serif;
  font-size: 14px;
  overflow: hidden;
  color: #333;
  text-overflow: ellipsis;
}

.c11 {
  font-size: 36px;
  line-height: 1.2;
  color: #333;
}

.c12 {
  width: 100%;
  margin-left: 6px;
}

.c2 {
  margin-bottom: 16px;
}

@media screen and (max-width:576px) {
  .c1 {
    width: 50%;
  }
}

@media screen and (min-width:576px) {
  .c1 {
    width: 33.33333333333333%;
  }
}

<section>
    <h2
      class="c0"
    >
      Links
    </h2>
    <div
      class="c1"
      rows="6"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          <div
            class="c4"
          >
            <i
              class="fab fa-twitter fa-fw c5"
            />
            <div
              class="c6"
            >
              <p
                class="sc-AxiKw sc-AxhCb c7"
              >
                Twitter 1
              </p>
              <p
                class="sc-AxiKw sc-AxhCb c8"
              >
                MikazukiFuyuno
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="c1"
      rows="6"
    >
      <div
        class="c2"
      >
        <a
          class="c9"
          href="/"
        >
          <div
            class="c10"
          >
            <i
              class="fal fa-envelope fa-fw c11"
            />
            <div
              class="c12"
            >
              <p
                class="sc-AxiKw sc-AxhCb c13"
              >
                Email 1
              </p>
              <p
                class="sc-AxiKw sc-AxhCb c14"
              >
                me@mochizuki.moe
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="c1"
      rows="6"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          <div
            class="c4"
          >
            <i
              class="fab fa-twitter fa-fw c5"
            />
            <div
              class="c6"
            >
              <p
                class="sc-AxiKw sc-AxhCb c7"
              >
                Twitter 2
              </p>
              <p
                class="sc-AxiKw sc-AxhCb c8"
              >
                MikazukiFuyuno
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="c1"
      rows="6"
    >
      <div
        class="c2"
      >
        <a
          class="c9"
          href="/"
        >
          <div
            class="c10"
          >
            <i
              class="fal fa-envelope fa-fw c11"
            />
            <div
              class="c12"
            >
              <p
                class="sc-AxiKw sc-AxhCb c13"
              >
                Email 2
              </p>
              <p
                class="sc-AxiKw sc-AxhCb c14"
              >
                me@mochizuki.moe
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="c1"
      rows="6"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/"
        >
          <div
            class="c4"
          >
            <i
              class="fab fa-twitter fa-fw c5"
            />
            <div
              class="c6"
            >
              <p
                class="sc-AxiKw sc-AxhCb c7"
              >
                Twitter 3
              </p>
              <p
                class="sc-AxiKw sc-AxhCb c8"
              >
                MikazukiFuyuno
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="c1"
      rows="6"
    >
      <div
        class="c2"
      >
        <a
          class="c9"
          href="/"
        >
          <div
            class="c10"
          >
            <i
              class="fal fa-envelope fa-fw c11"
            />
            <div
              class="c12"
            >
              <p
                class="sc-AxiKw sc-AxhCb c13"
              >
                Email 3
              </p>
              <p
                class="sc-AxiKw sc-AxhCb c14"
              >
                me@mochizuki.moe
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
